sdk/webpubsub/Microsoft.Azure.WebPubSub.Common/src/WebPubSubConnectionContext.cs

```diff
@@ -90,5 +92,39 @@ public WebPubSubConnectionContext(WebPubSubEventType eventType, string eventName
             States = states;
             Headers = headers;
         }
+
+        /// <summary>
+        /// Method to deserialize state to customized type.
+        /// </summary>
+        /// <typeparam name="T">Customized type.</typeparam>
+        /// <param name="stateKey">State key.</param>
+        /// <param name="value">Connection state mapped to the key.</param>
+        /// <returns>Returns true if is able to find the key and false means key doesn't exist or value type not able to convert.</returns>
+        public bool TryGetState<T>(string stateKey, out T value)
+        {
+            value = default(T);
+            if (States.TryGetValue(stateKey, out var stateValue))
+            {
+                if (stateValue is T val)
```

Comment:
What is the case where this can happen?

Reply:
T is JsonElement, or user build the WebPubSubConnectionContext themselves without putting origial data as JsonElement but actual some specific type. Then this won't result in unexpected default(T).